src/Components/Nav.js

```diff
@@ -0,0 +1,346 @@
+import React, { useEffect, useState } from "react";
+import { mainData } from "../MockData/MainData.js";
+import styled from "styled-components";
+
+const Nav = ({ display, visibility }) => {
+  const [toggle, setToggle] = useState(false);
+  const [searchText, setSearchText] = useState("");
+  const [data, setData] = useState(mainData.searchData);
+  const excludeColumns = ["id", "imgUrl"];
+
+  const onToggle = () => {
+    setToggle(toggle === false ? true : false);
+  };
+
+  const handleChange = (value) => {
+    setSearchText(value);
+    filterData(value);
+  };
+
+  const filterData = (value) => {
+    const lowercasedValue = value.toLowerCase().trim();
+    if (lowercasedValue === "") {
+      setData(mainData.searchData);
+    } else {
+      const filteredData = mainData.searchData.filter((item) => {
+        return Object.keys(item).some((key) =>
+          excludeColumns.includes(key)
+            ? false
+            : item[key].toString().toLowerCase().includes(lowercasedValue)
+        );
+      });
+      setData(filteredData);
+    }
+  };
+
+  const useScroll = () => {
+    const [scroll, setScroll] = useState(0);
+
+    const onScroll = () => {
+      setScroll({
+        y: window.scrollY,
+      });
+    };
```

Comment:
```suggestion
    const onScroll = () => {
      setScroll(window.scrollY);
    };
```
x, y 값을 다 쓴다면 저렇게 쓸텐데 이 경우는 y 값 뿐이라서 그냥 이렇게 쓰면 좋겠습니다